### PR TITLE
Added Convtranspose1d

### PR DESCRIPTION
**Pull request by jeffin07**
Added convtranspose1d and it's tests

### CODE REVIEW

**Review by imdeepmind — CHANGES_REQUESTED** [review]
There are some issues with this PR, please fix these issues only then I can merge this PR.

Issues in this PR:
- We've changed the layer structure code, please update your PR for that, You can check the latest code for the conv1d layer to see the new structure.
- We've moved to flake8 from PyLint and also added a new flake8 configuration file, so run flake8 on your branch and fix the code issues.

File: neuralpy/layers/convolutional/convtranspose1d.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        if bias is not None and not isinstance(bias, bool):
+            raise ValueError("Please provide a valid bias")
+
+        # Checking the dilation field, it is a optional field
+        if dilation is not None and not isinstance(
+                dilation, (int, tuple)):
+            raise ValueError("Please provide a valid dilation")
+
+        # Checking the name field, it is a optional field
+        if name is not None and not (isinstance(name, str) and name):
```
Comment (imdeepmind) [inline neuralpy/layers/convolutional/convtranspose1d.py:122]: You dont need to validate the name here. The `CustomLayer` class automatically does it for you. You can have a look at other layers for reference

File: neuralpy/layers/convolutional/convtranspose1d.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        self.__kernel_size = kernel_size
+
+        self.__input_shape = input_shape
+        self.__stride = stride
+        self.__padding = padding
+        self.__out_padding = output_padding
+        self.__groups = groups
+        self.__bias = bias
+        self.__dilation = dilation
+        self.__name = name
```
Comment (imdeepmind) [inline neuralpy/layers/convolutional/convtranspose1d.py:139]: You need to make a class field for name, the class you inherited automatically handles it for you

File: neuralpy/layers/convolutional/convtranspose1d.py [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+    def get_layer(self):
+        """
+        This method returns the details as dict of the layer.
+
+        This method is used by the NeuralPy Models, for building the models.
+        No need to call this method for using NeuralPy.
+        """
+        # Returning all the details of the layer
+        return self._get_layer_details(
```
Comment (imdeepmind) [inline neuralpy/layers/convolutional/convtranspose1d.py:224]: The return statement seems incorrect, please check

**Comment by imdeepmind** [comment]
@jeffin07 Seems like there is some linting issue with your code, can you check it once with `flake8`. Also, test coverage is around 86%, see.if you can improve the percentage number.

File: neuralpy/layers/convolutional/convtranspose1d.py (hunk 3 above)
Comment (jeffin07) [reply neuralpy/layers/convolutional/convtranspose1d.py:224]: @imdeepmind I can't figure out what I'm missing here. Can you please tell me what I'm missing here?

Comment (imdeepmind) [reply neuralpy/layers/convolutional/convtranspose1d.py:217]: Sorry, it seems fine. By mistake the method `__get_layer_details` and `_get_layer_details` are same, so got confused.

**Comment by imdeepmind** [comment]
@jeffin07 Need some time to check the calculation for the PR, anyways there are still some issues with flake here. Can you have a look at the issues? I'm detailing them below:

- E231 missing whitespace after ','
- E302 expected 2 blank lines, found 1
- E303 too many blank lines (4)
- W292 no newline at end of file
- W293 blank line contains whitespace
- W391 blank line at end of file

Run flake locally to see the details in more detail.

**Comment by jeffin07** [comment]
@imdeepmind I ran flake8 but it shows no errors or warnings.can you tell me in which file that you get these

**Comment by imdeepmind** [comment]
Where you're running the flake8? What command you're using to run it? Go to the root directory of the project and then run the following command `flake8 .`.

**Comment by jeffin07** [comment]
@imdeepmind I'm currently using ` python -m flake8 python file`. While using  `flake8 .` it shows some errors on file which I've not edited

**Comment by jeffin07** [comment]
@imdeepmind Any updates?

**Pull request closed without merging**